src/contexts/APIRouterContext/Provider.tsx

```diff
@@ -250,7 +267,7 @@ const APIRouterProvider: React.FC = ({ children }) => {
       }
 
       // TODO: should consider doing something with the return of this function call
-      await afterSecretCB.current(userSecretInput);
+      await afterSecretCB.current(userSecretInput, messageInput);
```

Comment:
just to be clear, message input is being ignored by the other _handle functions.

Reply:
perhaps a better way to do it is to check for a message to exist? i need to talk it over and think on it a bit more

Reply:
yea I'm aware it's being ignored in the others, i didn't really like the approach I had to take but I didn't see a better way to stitch in additional fields to consume

Reply:
see issue #115